pkg/controller/hibernation/hibernation_controller_test.go

```diff
@@ -418,19 +398,20 @@ func TestHibernateAfter(t *testing.T) {
 			result, err := reconciler.Reconcile(reconcile.Request{
 				NamespacedName: types.NamespacedName{Namespace: namespace, Name: cdName},
 			})
-			if test.expectRequeueAfter {
-				assert.True(t, result.RequeueAfter > 0)
-			} else {
-				assert.Equal(t, 0*time.Second, result.RequeueAfter)
+
+			// Need to do fuzzy requeue after matching
+			if assert.Nil(t, err) {
+				assert.True(t, test.expectRequeueAfter-10*time.Second < result.RequeueAfter &&
+					result.RequeueAfter < test.expectRequeueAfter+10*time.Second,
+					"RequeueAfter (%s) duration is not within 10s of expected (%s)",
+					result.RequeueAfter, test.expectRequeueAfter)
 			}
 
-			assert.Nil(t, err)
 			cd := &hivev1.ClusterDeployment{}
 			err = c.Get(context.TODO(), client.ObjectKey{Namespace: namespace, Name: cdName}, cd)
 			require.Nil(t, err)
 			assert.Equal(t, test.expectedPowerState, cd.Spec.PowerState)
```

Comment:
Add message to the assert.

Reply:
I will update but this should probably be discussed as a team, I am not in favor of requiring text for every assert when the code is self explanatory, it's just overly verbose and more churn in the review process.